website/pizzas/models.py

```diff
@@ -212,26 +234,17 @@ def clean(self):
     def save(self, *args, **kwargs):
         if not self.id and self.pizza_event.end_reminder:
             self.pizza_event.end_reminder.users.remove(self.member)
+        super().save(*args, **kwargs)
 
-        notes = (
-            f"Pizza order by {self.member_name} "
-            f"for {self.pizza_event.event.title_en}"
-        )
-        try:
+        if self.payment:
+            notes = (
+                f"Pizza order by {self.member_name} "
+                f"for {self.pizza_event.event.title_en}"
+            )
             self.payment.notes = notes
             self.payment.paid_by = self.member
             self.payment.amount = self.product.price
             self.payment.save()
-        except ObjectDoesNotExist:
-            self.payment = Payment.objects.create(
-                amount=self.product.price,
-                notes=notes,
-                paid_by=self.member,
-                topic=f"Pizzas {self.pizza_event.event.title_en} "
-                f'[{date(self.pizza_event.start, "Y-m-d")}]',
-            )
-
-        super().save(*args, **kwargs)
 
     def delete(self, using=None, keep_parents=False):
         if not self.id and self.pizza_event.end_reminder:
```

Comment:
The payment is removed here below when the Order is deleted, that should probably not happen. 
1. Because an order may be deleted even though it was processed
2. Because this happens in the model while it could be in a signal